src/internal/connector/onedrive/item_test.go

```diff
@@ -256,3 +258,158 @@ func (suite *ItemIntegrationSuite) TestDriveGetFolder() {
 		})
 	}
 }
+
+// Unit tests
+
+// mock
+type mockGetter struct {
+	GetFunc func(ctx context.Context, url string) (*http.Response, error)
+}
+
+func (m mockGetter) Get(
+	ctx context.Context,
+	url string,
+	headers map[string]string,
+) (*http.Response, error) {
+	return m.GetFunc(ctx, url)
+}
+
+type ItemUnitTestSuite struct {
+	tester.Suite
+}
+
+func TestItemUnitTestSuite(t *testing.T) {
+	suite.Run(t, &ItemUnitTestSuite{Suite: tester.NewUnitSuite(t)})
+}
+
+func (suite *ItemUnitTestSuite) TestDownloadItem() {
+	testRc := ioutil.NopCloser(bytes.NewReader([]byte("test")))
+
+	table := []struct {
+		name          string
+		itemFunc      func() models.DriveItemable
+		GetFunc       func(ctx context.Context, url string) (*http.Response, error)
+		errorExpected bool
+		rcExpected    bool
+		label         string
+	}{
+		{
+			name: "nil item",
+			itemFunc: func() models.DriveItemable {
+				return nil
+			},
+			GetFunc: func(ctx context.Context, url string) (*http.Response, error) {
+				return nil, nil
+			},
+			errorExpected: true,
+			rcExpected:    false,
+		},
+		{
+			name: "success",
+			itemFunc: func() models.DriveItemable {
+				di := newItem("test", false)
+				di.SetAdditionalData(map[string]interface{}{
+					"@microsoft.graph.downloadUrl": "https://example.com",
```

Comment:
could you add one more test that has the URL as the value for a field with key `"@content.downloadUrl"`? It can be the same as this test for all other checks and stuff